pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -894,55 +895,61 @@ func (r *Builder) PopulatorVolumes(vmRef ref.Ref, annotations map[string]string,
 	}
 
 	for _, image := range images {
-
 		if image.Status != string(ImageStatusActive) {
 			r.Log.Info("the image is not ready yet", "image", image.Name)
 			continue
 		}
-
-		originalVolumeDiskId := image.Name
-		if imageProperty, ok := image.Properties[forkliftPropertyOriginalVolumeID]; ok {
-			originalVolumeDiskId = imageProperty.(string)
-		}
-
-		_, err = r.getVolumePopulator(image.Name)
+		var pvc *core.PersistentVolumeClaim
+		pvc, err = r.ensureVolumePopulator(workload, &image, annotations, secretName)
 		if err != nil {
-			if !k8serr.IsNotFound(err) {
-				err = liberr.Wrap(err)
+			if !k8serr.IsAlreadyExists(err) {
+				err = liberr.Wrap(err, "couldn't build the PVC",
+					"image", image.Name)
 				return
 			}
+			err = nil
+			continue
```

Comment:
shouldn't we proceed to `pvcNames = append(pvcNames, pvc.Name)` in this case?

Reply:
outdated, please check the latest push